docs: Bigtable change streams song rank example/tutorial (#8254)
GoogleCloudPlatform/java-docs-samples: bigtable/beam/change-streams/README.md

## Hello World

> This example outputs a message to standard output when a change is made to Bigtable. 

Reviewer: ```suggestion
This example writes a message to standard output when a change is made to Bigtable. 
```

1. Create a table with a change streams enabled

   ```sh
   gcloud bigtable instances tables create $TABLE_ID \
    --column-families=cf1 --change-stream-retention-period=7d \

>    --instance=INSTANCE_ID --project=$GOOGLE_CLOUD_PROJECT 

Author: ah, I thought they had added it already

1. Run the test

   ```sh
   mvn clean test -Dtest=ChangeStreamsHelloWorldTest
   ```

## Song rank

> This example keeps track of songs listened to and gets the top 5 songs over a period.

Reviewer: ```suggestion
This example keeps track of songs listened to and gets the top 5 songs over a period of time.
```

The top 5 songs are outputted to standard out and files which can be local or on Google Cloud Storage.

1. Create a Bigtable instance or use an existing one

1. Set up the environment variables

    ```sh
    GOOGLE_CLOUD_PROJECT=your-project-id
    BIGTABLE_PROJECT=your-project-id
    INSTANCE_ID=your-instance-id
    TABLE_ID=song-rank
    REGION=us-central1
    OUTPUT_LOCATION=gs://your-bucket-id  # Exclude the gs:// to save locally 
    ```

1. Create a table with a change streams enabled

   ```sh
   gcloud alpha bigtable instances tables create $TABLE_ID \
    --column-families=cf --change-stream-retention-period=7d \

>    --instance=$BIGTABLE_TESTING_INSTANCE --project=$GOOGLE_CLOUD_PROJECT 

Reviewer 2: ```suggestion
   --instance=$INSTANCE_ID --project=$GOOGLE_CLOUD_PROJECT 
```

inconsistent env var

   ```

1. Run command to start the pipeline

>     ```sh
>     mvn compile exec:java -Dexec.mainClass=SongRank \
>     "-Dexec.args=--project=$GOOGLE_CLOUD_PROJECT --bigtableProjectId=$BIGTABLE_PROJECT \
>     --bigtableInstanceId=$INSTANCE_ID --bigtableTableId=$TABLE_ID --outputLocation=$OUTPUT_LOCATION \
>     --runner=dataflow --region=$REGION --experiments=use_runner_v2"
>     ```

Reviewer 2: ```suggestion
    ```sh
    mvn compile exec:java -Dexec.mainClass=SongRank \
    "-Dexec.args=--project=$GOOGLE_CLOUD_PROJECT --bigtableProjectId=$BIGTABLE_PROJECT \
    --bigtableInstanceId=$BIGTABLE_TESTING_INSTANCE --bigtableTableId=$TABLE_ID --outputLocation=$OUTPUT_LOCATION \
    --runner=dataflow --region=$REGION --experiments=use_runner_v2"
    ```
```

inconsistent env var

1. Stream some data which contains song listens for various users

```sh
cbt -instance=$BIGTABLE_INSTANCE_ID -project=$GOOGLE_CLOUD_PROJECT import\
$TABLE_ID song-rank-data.csv  column-family=cf batch-size=1
```

1. Observe the output and see the most popular songs.

1. Stop your Dataflow job to avoid incurring any costs.

### Testing

> 

Author: You should be able to use any version of java for the tests

1. Set environment variables

   ```sh
   GOOGLE_CLOUD_PROJECT="project-id"
   BIGTABLE_TESTING_INSTANCE="instance-id"
   ```

1. Run the command to create a test table with change streams enabled

   ```sh
   gcloud alpha bigtable instances tables create song-rank-test \
    --column-families=cf --change-stream-retention-period=7d \
   --instance=$BIGTABLE_TESTING_INSTANCE --project=$GOOGLE_CLOUD_PROJECT
   ```

1. Run the test

   ```sh
   mvn clean test -Dtest=SongRankTest
   ```

## Running locally

To run either program on your local machine, you can use the direct Beam runner
by

> setting `--runner=direct`. If you're running it locally, you don't need the
> `--project` or `--region` parameters.

Reviewer 2: ```suggestion
setting `--runner=DirectRunner` (also the default if not specified). If you're running it locally, you don't need the
`--project` or `--region` parameters.
```
